### DISCUSSION TITLE
How to obtain customer ID and API key

### DISCUSSION THREAD

**Discussion by t-saggis** [robbinjanssen/home-assistant-ojmicroline-thermostat#56] [Q&A] [answered]
Hi
How do I find my customer ID and API key? When using 99 and f219aab4-9ac0-4343-8422-b72203e2fac9  + username/password from my APP I only get "Invalid authentication"

**Comment by robbinjanssen** [comment]
It's kind of complicated and includes modifying the APK of an android app. Which app do you use on your phone?

**Comment by t-saggis** [comment]
I'm using the Microtemp app on an iPhone. However, if required I can make an installation on an Android and perform necessary modifications but it would probably require some instructions. :) And the easiest would of course just try out a set of combinations of custuomer ID and API key if we know that some of those combinations will work.

**Reply by robbinjanssen** [reply, marked as answer]
Did you also set the correct host and customer ID? Here `owd5-mh015-app.ojelectronics.com` and `111` is used: https://community.home-assistant.io/t/mwd5-wifi-thermostat-oj-electronics-microtemp/445601

To "sniff" the API key you need to modify and emulate the app, and the decipher the SSL traffic between your host and the emulator.. It's kind of a hassle, an example can be seen here: 
- https://medium.com/@roy2477/decrypting-android-app-ssl-traffic-c480e2143743
- https://nerds.ontruck.com/capturing-ssl-traffic-from-an-android-apk-879e6cce9aae

**Reply by robbinjanssen** [reply]
Btw, I use the same API key with customer ID 99 and another host. But I use the `SWATT` app on my phone
